src/ZoneServer/Buffs/Handlers/EnchantEarth_Buff.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Melia.Shared.Tos.Const;
+using Melia.Zone.Buffs.Base;
+using Melia.Zone.Scripting;
+using Melia.Zone.World.Actors.Characters;
+
+namespace Melia.Zone.Buffs.Handlers
+{
+	/// <summary>
+	/// Handle for the Enchant Earth Buff, which increases the target's block penetration
+	/// </summary>
+	[BuffHandler(BuffId.EnchantEarth_Buff)]
+	public class EnchantEarth_Buff : BuffHandler
+	{
+		private float blockPenetrationBonus = 0;
+
+		public override void OnStart(Buff buff)
+		{
+			var target = buff.Target as Character;
+			var SCR_Get_Character_MAXPATK = ScriptableFunctions.Character.Get("SCR_Get_Character_MAXPATK");
+
+			if (target != null)
+			{
+				var caster = buff.Caster as Character;
+
+				// Apply penality when the CASTER Max Physical Attack is lower than the TARGET Max Physical Attack
+				// TODO: Find out the exacly value of the penality (We are applying 50%)
+				var penalityValue = SCR_Get_Character_MAXPATK(caster) < SCR_Get_Character_MAXPATK(target) ? 0.5f : 1f;
+
+				// Based on in game values (tooltip)
```

Comment:
It's very possible that I'm missing something, but I don't quite understand your formula in relation to the in-game description. Enchant Earth is supposed to raise your Block Penetration by X% per skill level, where X appears to be `6 + (level -1)`. Shouldn't this buff be as simple as adding that value to `BLK_BREAK_RATE_BM`?

Reply:
`var penalityValue = SCR_Get_Character_MAXPATK(caster) < SCR_Get_Character_MAXPATK(target) ? 0.5f : 1f;`

This part is related to:

`The final efficiency is reduced when the highest Physical ATK of the caster is lower than the highest ATK of the target.`